### PR TITLE
fix: social urls from config

### PR DESCRIPTION
**Pull request by jeff350**
Addresses #4511  by using required config option `SOCIAL_AUTH_URL_NAMESPACE`

### CODE REVIEW

**Review by dkliban — CHANGES_REQUESTED** [review]
please add a changelog file to the CHANGES directory. I would say that this is a new feature. So it should be called `4511.feature`.

Your commit message needs to also state 'fixes: #4511' or 'closes: #4511'

**Comment by jeff350** [comment]
@dkliban requested updates have been made, please let me know if I need to change all commit messages to 'fixes: #4511'  or if just the latest is sufficient.

**Comment by dkliban** [comment]
@jeff350 please squash the changes into a single commit

**Comment by jeff350** [comment]
@dkliban all commits have been squashed into a single commit, looks like this should be good to go when all tests pass.

**Comment by dkliban** [comment]
@jeff350 The CI is showing the following error:

` AttributeError: 'HookableSettings' object has no attribute 'SOCIAL_AUTH_URL_NAMESPACE'`

You need to ensure that the setting is present before checking the value.

**Comment by jeff350** [comment]
@dkliban I missed that case... I am starting to wonder if checking that specific setting directly may not the best approach. A possible alternative would be to check if social_django is in the list of installed apps, that maybe a cleaner way to do this.

 do you have an option on what would be preferred?

**Comment by dkliban** [comment]
@jeff350 You should check that `django_social` is in the installed apps. If it is installed, you should then check that `SOCIAL_AUTH_URL_NAMESPACE` is set. Then the value of `SOCIAL_AUTH_URL_NAMESPACE` should be used as the value of `namespace` when adding the You also want to make sure that the value of `namespace` in the urlpattern being added.

**Comment by jeff350** [comment]
@dkliban This should be fixed now. I also added the SOCIAL_AUTH_URL_NAMESPACE as a default in the settings.py file as I understand that will always need to be a custom namespace and 'social' is what django_social always recommends to use. 

This also helped to further simplify the docs.

File: pulpcore/app/urls.py [hunk 1]
```diff
@@ -203,6 +203,10 @@ class NoSchema(p.callback.cls):
         docs_and_status_no_schema.append(path(str(p.pattern), view, name=name))
     urlpatterns.append(path(API_ROOT, include(docs_and_status_no_schema)))
 
+if "social_django" in settings.INSTALLED_APPS:
+    if settings.SOCIAL_AUTH_URL_NAMESPACE == "social":
```
Comment (dkliban) [inline pulpcore/app/urls.py:207]: I don't believe that we want to only include the urlpattern when the SOCIAL_AUTH_URL_NAMESPACE is 'social'. If 'social_django' is installed we want to include the urlpattern with the `namespace` specified by `SOCIAL_AUTH_URL_NAMESPACE`.  So like this:

```python
if "social_django" in settings.INSTALLED_APPS:
    urlpatterns.append(path("", include("social_django.urls", namespace=settings.SOCIAL_AUTH_URL_NAMESPACE)))
```

Comment (jeff350) [reply pulpcore/app/urls.py]: Change has been mande

Comment (mdellweg) [reply pulpcore/app/urls.py]: Thank you. I was about to request the exact same.

File: pulpcore/app/settings.py [hunk 2]
```diff
@@ -202,6 +202,9 @@
 REDIS_SSL = False
 REDIS_SSL_CA_CERTS = None
 
+# Social_django settings
+SOCIAL_AUTH_URL_NAMESPACE = "social"
+
```
Comment (mdellweg) [inline pulpcore/app/settings.py:207]: Please add a section to the settings part of the docs, too.

**Review by mdellweg — CHANGES_REQUESTED** [review]

File: docs/authentication/keycloak.rst [hunk 3]
```diff
@@ -52,20 +52,7 @@ Enable general python social integration with the following steps:
     SOCIAL_AUTH_POSTGRES_JSONFIELD = True
     SOCIAL_AUTH_JSONFIELD_CUSTOM = 'django.db.models.JSONField'
 
-
-4. In order to use the Keycloak OIDC capabilities, you must add a URL namespace::
-
-    SOCIAL_AUTH_URL_NAMESPACE = 'social'
-
```
Comment (mdellweg) [inline docs/authentication/keycloak.rst:59]: By now providing a default value, can we still keep this step as an optional one?

Comment (jeff350) [reply docs/authentication/keycloak.rst:59]: My understanding is that this is not a setting anyone will want to change. Reading the Django docs (https://docs.djangoproject.com/en/4.2/topics/http/urls/#url-namespaces) the namespace is designed for reserving url patterns. I would equate django_social having the namespace of 'social' to their example of the django admin having a namespace of 'admin'.

additionally  django_social also recommends setting this to 'social' if a custom namespace is going to be used.  https://python-social-auth.readthedocs.io/en/latest/configuration/django.html#urls-entries

File: pulpcore/app/settings.py (hunk 2 above)
Comment (jeff350) [reply pulpcore/app/settings.py:207]: If this is not something that is intended for users to change does it need a section in the settings documentation? looking at the documentation it seems that the settings there are ones that are intended fore users to change as needed. 

See for context on why this does not need to be a user configurable setting https://github.com/pulp/pulpcore/pull/4512#discussion_r1341971011

File: docs/authentication/keycloak.rst (hunk 3 above)
Comment (mdellweg) [reply docs/authentication/keycloak.rst:59]: > additionally django_social also recommends setting this to 'social' if a custom namespace is going to be used. https://python-social-auth.readthedocs.io/en/latest/configuration/django.html#urls-entries

Actually, i read it the other way around. You only need to set it if it's _not_ the default (and their example is not helping much). Having said that, the url-namespace is not changing the included paths. So, I think we are fine always using "social".

File: docs/authentication/keycloak.rst [hunk 4]
```diff
@@ -52,20 +52,7 @@ Enable general python social integration with the following steps:
     SOCIAL_AUTH_POSTGRES_JSONFIELD = True
     SOCIAL_AUTH_JSONFIELD_CUSTOM = 'django.db.models.JSONField'
```
Comment (mdellweg) [inline docs/authentication/keycloak.rst:53]: We are always using Postgres. You can move these settings too and remove the step from the docs.

Comment (jeff350) [reply docs/authentication/keycloak.rst]: Requested changes have been made

File: pulpcore/app/settings.py [hunk 5]
```diff
@@ -202,6 +202,11 @@
 REDIS_SSL = False
 REDIS_SSL_CA_CERTS = None
 
+# Social_django settings
+SOCIAL_AUTH_URL_NAMESPACE = "social"
+SOCIAL_AUTH_POSTGRES_JSONFIELD = True
+SOCIAL_AUTH_JSONFIELD_CUSTOM = 'django.db.models.JSONField'
```
Comment (dkliban) [inline pulpcore/app/settings.py:208]: black is complaining about use of single quotes. Please use " instead of '.

Comment (jeff350) [reply pulpcore/app/settings.py]: fixed

**Review by mdellweg — APPROVED** [review]

**Review by dkliban — APPROVED** [review]

**Pull request merged**